### PR TITLE
PHPCS support

### PR DESCRIPTION
**Pull request by paulschreiber**
* add phpcs.xml
* update composer.json
* gitignore vendor folder
* run PHPCBF to fix spacing problems

### CODE REVIEW

**Comment by paulschreiber** [comment]
Tests are passing. Failure was a travis error for WP4.8:
>The "https://getcomposer.org/download/1.6.5/composer.phar.sig" file could not
>be downloaded: failed to open stream: Connection timed out

File: phpcs.xml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0"?>
+<ruleset name="WordPress Coding Standards for Co-Authors Plus">
+
+	<rule ref="WordPress-VIP" />
+	<rule ref="WordPress-Docs" />
+	<rule ref="WordPress-Extra" />
+
+	<arg value="s"/>
```
Comment (GaryJones) [inline phpcs.xml]: Consider adding flags to show progress, use shorter basepath, adding colors etc. The exact args available depends on what PHPCS version is installed, and that depends on what minimum PHP version should be supported.

File: phpcs.xml [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+	<arg name="extensions" value="php"/>
+	<file>.</file>
+
+	<exclude-pattern>*/node_modules/*</exclude-pattern>
+	<exclude-pattern>*/vendor/*</exclude-pattern>
+</ruleset>
```
Comment (GaryJones) [inline phpcs.xml]: By enabling some custom properties (see https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards/wiki/Customizable-sniff-properties#properties-strongly-recommended-to-be-set) then extra checks, such as textdomains and deprecations will also be run.

**Comment by GaryJones** [comment]
Some of the config changes have been handled more recently, and some of the changes are now redundant. There is a lot of PHPCS to tidy up, but it's going to be needed to be done in smaller chunks.

**Pull request closed without merging**